pywbem_mock/_namespaceprovider.py

```diff
@@ -0,0 +1,290 @@
+                'CIM_ObjectManager' \
+                or new_instance['SystemName'] != SYSTEMNAME \
+                or new_instance['SystemCreationClassName'] != \
+                    SYSTEMCREATIONCLASSNAME:
+
+                raise CIMError(
+                    CIM_ERR_INVALID_PARAMETER,
+                    _format("Namespace creation via CreateInstance: "
+                            "Invalid property for CIM_Instance {0!A}",
+                            new_instance))
+
+        except KeyError as ke:
+            raise CIMError(
+                CIM_ERR_INVALID_PARAMETER,
+                _format("Namespace creation via CreateInstance: "
+                        "Missing property for CIM_Instance {0!A}. Exception "
+                        "{1!A}", new_instance, ke))
+
+        if new_namespace not in self.cimrepository.namespaces:
+            self.cimrepository.add_namespace(new_namespace)
+
+        # Pass NewInstance to the default CreateInstance
+        # Default CreateInstance creates path, validates all key properties
+        # in instance and inserts instance into the repository.
+        return super(CIMNamespaceProvider, self).CreateInstance(namespace,
+                                                                new_instance)
+
+    def ModifyInstance(self, ModifiedInstance,
+                       IncludeQualifiers=None, PropertyList=None):
+        """
+        Modification of CIM_Namespace instance not allowed
+        """
+        raise CIMError(
+            CIM_ERR_NOT_SUPPORTED,
+            _format("Modification of CIM_Namespace {0!A} not implemented. ",
+                    ModifiedInstance.classname))
+
+    def DeleteInstance(self, InstanceName):
+        """
+        Delete the namespace defined by the Name property in InstanceName
+        and remove the instance defined by InstanceName
+        """
+        remove_namespace = InstanceName.keybindings['Name']
+
+        # Reflect the namespace deletion in the CIM repository
+        # This will check the namespace for being empty.
+        self.remove_namespace(remove_namespace)
+
+        # Delete the instance from the CIM repository
+        instance_store = self.get_instance_store(InstanceName.namespace)
+        instance_store.delete(InstanceName)
+
+    @staticmethod
+    def install_provider(conn, interop_namespace, pragma_file,
+                         verbose=None):
+        """
+        FakedWBEMConnection user method to install the namespace provider in
+        the interop namespace where the proposed interop_namespace is defined
+        by the parameter interop_namespace
+
+        Because this provider require a set of classes from the
+        DMTF schema, the parameters to install the schema are required.
+
+        This method should only be called once at the creation of the
+        mock environment.
+
+        Parameters:
+
+          interop_namespace  (:term:`string`):
+            The interop namespace defined for this environment.
```

Comment:
It is the interop namespace for which this provider should be registered.

Reply:
DONE.. Added text